src/hooks/useLookupNotification.ts

```diff
@@ -24,22 +28,37 @@ export default function useLookupNotification(value: unknown) {
   // and have it aborted if the user clicks the lookup button again.
   const [lookupCount, setLookupCount] = React.useState(0)
 
+  // We use a ref here to keep the abort controller outside of the useEffect
+  // below. This way we can abort the lookup if the value changes while the
+  // lookup is running.
+  const abortControllerRef = React.useRef<AbortController | null>(null)
+
   React.useEffect(() => {
     if (lookupCount === 0) {
       return
     }
-    const abortController = new AbortController()
-    onLookup(value, abortController.signal)
-    return () => {
-      abortController.abort()
-    }
+    abortControllerRef.current = new AbortController()
+    onLookup({
+      newValue: value,
+      abortController: abortControllerRef.current,
+      continueLookupOnAbort: false,
+    })
     // Wants to use "onLookup" and "value" as dependencies,
     // however, these will change on any change made on any
     // element. Checking if "lookupCount" has changed is enough
     // to trigger a lookup when the correct dependencies change
     // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
now that I'm a react docs aficionado,  it would be cool to write this without the disable, but equally this has worked for a long time. Up to you (not a challenge)

Reply:
I am ashamed to say this has already cost me a few hours sleep and I can't work out a way to do this without a bunch of refs (I prefer this).